crates/txtx-cli/src/cli/runbooks/mod.rs

```diff
@@ -129,6 +126,26 @@ pub fn display_snapshot_diffing(
         println!("These actions will be re-executed in the next run.\n");
     }
 
+    if !has_critical_changes && !has_unexecuted && consolidated_changes.new_plans_to_add.is_empty()
+    {
+        println!("{} Latest snapshot in sync with latest runbook updates\n", green!("✓"));
+
+        // todo: if we had no critical changes, but there are some synthesized changes, this means the
+        // synthesized changes were non-critical. we should consider updating our runbook state file
+        // to match this new state, but not actually execute any actions.
+        // my concern here is that the state file could be used as a source of truth. For example, if I change something like
+        // the RPC API URL, that's non-tainting (because if I change the url, that doesn't mean I want to execute
+        // again, I just want a different RPC in the future) but it would be misleading to update the state file to reflect
+        // the new RPC - that's not what we executed in the past.
+        // issue: https://linear.app/txtx/issue/TXTX-265/update-statefile-on-non-critical-changes
+        // if we end up doing this:
+        // if !synthesized_changes.is_empty() {
+        // .. update state file
+        // }
+
```

Comment:
Thanks for unpacking the rationale, this code site is getting more and more subtile and complex.